### PR TITLE
Popout notes in window

### LINKED ISSUE #1076: Ability to pop out a Note to a Floating Window with More Flexible Resizing
### An example of a desired size:
image

### This should be achievable by a button to pop out the note out of the UI like many minimal Notes app do.

### PR DESCRIPTION
**Pull request by charliescheer**
### Fix
Currently you cannot open up notes in Simplenote Mac in their own window.  It is a fairly standard practice for writing apps like Simplenote that you would be able to pop a note out into a new window and edit it there.  This allows for editing multiple notes at the same time. 

In this PR I have added multiple window support for notes.  If a note is popped out into a new window we do not allow for that note to be edited in the main window will the window closes.  This is to make sure that we don't end up with multiple editors trying to access the same note at the same time.

fixes #1076 

### Test
1. Build SN Mac and go to the notes list
2. double click on any note

 - [ ] Confirm that the selected note opens in a new window
 - [ ] confirm that the note is closed in the main window
 - [ ] confirm you can select other notes in the main window
 - [ ] confirm that if you try to select the first note again (the one in the extra window) it opens the window and not in the main window
 - [ ] confirm once you close the extra window you are able to edit the note in the main window again

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in 9f3cfe with:
> 
> > Added multiple window support for editing notes

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by roundhill — CHANGES_REQUESTED** [review]
This is so cool! I found a bug or at least something I think should be fixed. When you open up a note in the window, make changes, then close it, you won't see the changes to the note in the list view or the editor in the right-pane. We should maybe refresh the notes in the main window after it gets focus again?

**Review by roundhill — APPROVED** [review]
Tested well and code LGTM!

File: Simplenote/NoteWindowManager.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        super.init(window: window)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class NoteWindowManager: NSObject {
+    var controllers = Set<NSWindowController>()
```
Comment (jleandroperez) [inline Simplenote/NoteWindowManager.swift]:
```suggestion
    private(set) var controllers = Set<NSWindowController>()
```

File: Simplenote/SimplenoteAppDelegate+Swift.swift [hunk 2]
```diff
@@ -511,6 +521,22 @@ extension SimplenoteAppDelegate {
...
+    }
+
+    var noteWindows: [NoteWindow] {
+        windows.compactMap({ $0 as? NoteWindow })
+    }
+
+    func noteWindow(for note: Note) -> NoteWindow? {
+        noteWindows.first(where: { $0.selectedNoteID == note.simperiumKey })
+    }
+}
```
Comment (jleandroperez) [inline Simplenote/SimplenoteAppDelegate+Swift.swift]: Do we still need these?

**Review by jleandroperez — APPROVED** [review]
Working solid on my end. No blockers, feel free to :shipit: Charile!!

**Pull request merged**